x-pack/plugins/security_solution/server/usage/detections/detections.test.ts

```diff
@@ -106,8 +106,79 @@ describe('Detections Usage and Metrics', () => {
     });
   });
 
+  describe('getDetectionRuleMetrics()', () => {
```

Comment:
In terms of testing this new logic with mock data, I would recommend you take a look at [`getAlertMock`](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/lib/detection_engine/routes/__mocks__/request_responses.ts#L348) and the various ways that we're using that 👍